pkg/network/netbinding/netbinding.go

```diff
@@ -33,6 +33,8 @@ import (
 )
 
 func NetBindingPluginSidecarList(vmi *v1.VirtualMachineInstance, config *v1.KubeVirtConfiguration, recorder k8srecord.EventRecorder) (hooks.HookSidecarList, error) {
```

Comment:
It feels wrong that it is [art of the SideCarList creation. The plugin may have only CNI or just a domainAttachment. I think the validation and the event should be separate.

Reply:
Ahh you are right! I had the same feeling but could not give a good reasoning.. You gave a really good reason why it is not a fit!

Reply:
I extracted the validation logic to its own function , please take another look.